phpBB/includes/functions_upload.php

```diff
@@ -899,6 +909,18 @@ function remote_upload($upload_url, \phpbb\mimetype\guesser $mimetype_guesser =
 						$file = new fileerror($user->lang[$this->error_prefix . 'URL_NOT_FOUND']);
 						return $file;
 					}
+					else if (stripos($line, 'location: ') !== false)
+					{
+						//there is a redirect, follow up to 5
+						if ($redirect_count > 5)
+						{
+							$file = new fileerror($user->lang[$this->error_prefix . 'URL_NOT_FOUND']);
+							return $file;
+						}
+
+						$upload_url = rtrim(str_replace('location: ', '', strtolower($line)));
+						return $this->remote_upload($upload_url, $mimetype_guesser, ++$redirect_count);
```

Comment:
Generally I agree. But there is a fair amount of logic would either need to be repeated or wrapping the whole lot in a loop. Personally I think the recursion is the best way here